src/TSID/src/QPTSID.cpp

```diff
@@ -535,3 +535,19 @@ bool QPTSID::isOutputValid() const
 {
     return m_pimpl->isValid;
 }
+
+std::string QPTSID::toString() const
+{
+    std::ostringstream oss;
+
+    oss << "====== QPTSID class ======" << std::endl
+        << "The optimization problem is composed by the following tasks:" << std::endl;
+    for (const auto& [name, task] : m_pimpl->tasks)
+    {
+        oss << "\t - " << name << ": " << task.task->getDescription() << " Priority: "
+            << task.priority << std::endl;
+    }
+    oss << "Note: The lower is the integer associated to the priority, the higher is the priority.";
```

Comment:
It is intended that the string is not ended by a newline even if it contains internal several newlines?

Reply:
Thank you @traversaro. I forgot to add it

Reply:
Ack, in any case in both ways it was ok for me.